webapp/src/ts/training-card.guard.provider.ts

```diff
@@ -0,0 +1,47 @@
+import { ActivatedRouteSnapshot, CanDeactivate, RouterStateSnapshot } from '@angular/router';
+import { Injectable } from '@angular/core';
+import { Store } from '@ngrx/store';
+
+import { GlobalActions } from '@mm-actions/global';
+import { Selectors } from '@mm-selectors/index';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class TrainingCardDeactivationGuardProvider implements CanDeactivate<any> {
+  private globalActions;
+  private trainingCardOpen = false;
+  private formId: string | null = null;
+
+  constructor(private store: Store) {
+    this.globalActions = new GlobalActions(store);
```

Comment:
The IDE marks the constructor's parameter `private store: Store)` as unused. The warning goes away by doing this: 
```suggestion
    this.globalActions = new GlobalActions(this.store);
```

Reply:
The magic of Webstorm. Vscode does not complain :(. 

How about the sonar complain that [router](https://github.com/medic/cht-core/pull/9512/files#diff-94d07e089bc01aba387ccbb407e1b7ed08b021c77672da72e2511b1a92b4cef9:~:text=private%20router%3A%20Router%2C) should be marked as readonly. Is that correct ? I was checking other parts of the code and I didn't see that.